core/shared/src/main/scala/org/http4s/internal/Logger.scala

```diff
@@ -40,7 +41,7 @@ object Logger {
   def defaultLogBody[F[_]: Concurrent](
       message: Message[F]
   )(logBody: Boolean): Option[F[String]] =
-    if (logBody) {
+    if (logBody && message.entity != Entity.Empty) {
```

Comment:
I wonder if `.isEmpty` / `.nonEmpty` type method would be useful.

Reply:
Yep, I've seen twice this checking for empty in our codebase. I'd iterate on this separately if no one isn't against it.